pkg/controller/certificates/issuing/issuing_controller.go

```diff
@@ -237,17 +248,6 @@ func (c *controller) ProcessItem(ctx context.Context, key string) error {
 		return c.failIssueCertificate(ctx, log, crt, req)
 	}
 
-	// Verify the CSR options match what is requested in certificate.spec.
-	// If there are violations in the spec, then the requestmanager will handle this.
-	requestViolations, err := certificates.RequestMatchesSpec(req, crt.Spec)
-	if err != nil {
-		return err
-	}
-	if len(requestViolations) > 0 {
-		log.V(logf.DebugLevel).Info("CertificateRequest does not match Certificate, waiting for keymanager controller")
-		return nil
-	}
-
```

Comment:
I had to move this block up because the issuing controller was checking the certificaterequest's failing condition (and updating the certificate) before checking if this certificaterequest was matching the certificate ([see investigation notes](https://github.com/jetstack/cert-manager/pull/3444#issuecomment-727598771)).

Due to this bug, the issuing controller would pick up an outdated certificaterequest that would be `Failing`, meaning that the newest certificaterequest would never be reconciled since the certificate would become `Failing` and `Issuing = False`.

Reply:
I moved this change to another PR with a proper unit test: #3478